src/cli/self_update.rs

```diff
@@ -1074,10 +1080,10 @@ pub(crate) fn uninstall(no_prompt: bool) -> Result<utils::ExitCode> {
 /// (and on windows this process will not be running to do it),
 /// rustup-init is stored in `CARGO_HOME`/bin, and then deleted next
 /// time rustup runs.
-pub(crate) fn update(cfg: &Cfg) -> Result<utils::ExitCode> {
-    common::warn_if_host_is_emulated();
-
+pub(crate) async fn update(cfg: &Cfg) -> Result<utils::ExitCode> {
     use common::SelfUpdatePermission::*;
+
+    common::warn_if_host_is_emulated();
```

Comment:
For a large change like this, I'd prefer not to make unrelated changes.

Reply:
sorry, what unrelated change?

Reply:
ah yes. Do you want me to back it out?

Reply:
Yes, I think that is preferrable.

Reply:
@djc Can we mark this as resolved now?